ui/src/app/shared/header/header.component.ts

```diff
@@ -174,8 +174,13 @@ export class HeaderComponent implements OnInit, AfterViewInit, AfterViewChecked,
   }
 
   removeChip(chipKey: string): void {
-    this.chips.delete(chipKey);
-    this.search();
+    if (this.chips.get(chipKey).valueOf()) {
+      this.chips.delete(chipKey);
+      this.search();
+    }
+    else {
+      this.chips.delete(chipKey);
```

Comment:
nit: move out of the else:
```
this.chips.delete(chipKey);
if (this.chips.get(chipKey).valueOf()) {
  this.search();
}
```

Reply:
If the chip has been deleted, `this.chips.get(chipKey)` returns an error; I can assign it to a variable and then use that in the if statement, if you think that would be cleaner.